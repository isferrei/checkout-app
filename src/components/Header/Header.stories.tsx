import type { Meta, StoryObj } from "@storybook/react";

import { Header } from "./Header";

const meta = {
  title: "Example/Header",
  component: Header,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta;

export default meta;
type Story = StoryObj;

export const Default = {
  args: {},
} satisfies Story;
