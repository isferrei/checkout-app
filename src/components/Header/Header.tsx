import Link from "next/link";

type HeaderProps = {
  href: string;
};

export function Header(props: HeaderProps) {
  return (
    <header className="flex flex-row w-full pt-[2.4rem] px-[4.2rem] items-center">
      <Link href={props.href}>
        <img src="/images/arrow.svg" alt="back arrow" />
      </Link>
      <nav className="mx-auto">
        <img src="/images/logo.svg" alt="whitebook" />
      </nav>
    </header>
  );
}
