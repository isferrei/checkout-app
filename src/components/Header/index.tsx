import { ActiveLink } from "../Activelink";

export function Header() {
    return (
        <header className="flex flex-row w-full pt-[2.4rem] px-[4.2rem] items-center">
            <ActiveLink href="/" prefetch activeClassName="">
                <img src="/images/arrow.svg" alt="back arrow" />
            </ActiveLink>
            <nav className="mx-auto">
                <img src="/images/logo.svg" alt="whitebook" />
            </nav>
        </header>
    );
}
