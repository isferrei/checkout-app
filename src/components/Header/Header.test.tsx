import { render, screen } from "@testing-library/react";
import { Header } from "./Header";

jest.mock("next/router", () => ({
  useRouter: () => ({
    pathname: "/",
  }),
}));

describe("Header component", () => {
  describe("Render method", () => {
    it("should have a href", async () => {
      render(<Header href="/" />);

      const headerElement = screen.getByRole("link");
      expect(headerElement).toBeInTheDocument();
    });
  });
});
