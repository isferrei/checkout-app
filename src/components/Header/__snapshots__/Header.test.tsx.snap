// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Render method should match the snapshot 1`] = `
<div>
  <header
    class="flex flex-row w-full pt-[2.4rem] px-[4.2rem] items-center"
  >
    <a
      href="/"
    >
      <img
        alt="back arrow"
        src="/assets/arrow.svg"
      />
    </a>
    <nav
      class="mx-auto"
    >
      <img
        alt="whitebook"
        src="/assets/logo.svg"
      />
    </nav>
  </header>
</div>
`;
